net/isoping/isoping.go

```diff
@@ -0,0 +1,244 @@
+	Id         uint32 // Id is a sequential packet id number
+	Txtime     uint32 // Txtime is the transmitter's monotonic time when pkt was sent
+	Clockdiff  uint32 // Clockdiff is an estimate of (transmitter's clk) - (receiver's clk)
+	UsecPerPkt uint32 // Usec_per_pkt microseconds of delay between packets
+	NumLost    uint32 // Num_lost is the number of pkts transmitter expected to get but didn't
+	FirstAck   uint32 // First_ack is the starting index in acks[] circular buffer
+	Acks       [64]struct {
+		// txtime==0 for empty elements in this array.
+		Id     uint32 // Id field from a received packet
+		Rxtime uint32 // Rxtime is a receiver's monotonic time when pkt arrived
+	}
+}
+
+type Isoping struct {
+	ClockStartTime time.Time    // ClockStartTime is the time the program starts
+	IsServer       bool         // IsServer distinguishes if we are a server or client
+	Conn           *net.UDPConn // Conn is either the server or client's connection
+	Tx             Packet       // Tx is a Packet that will be sent
+	Rx             Packet       // Rx is a Packet that will be received
+	LastAckInfo    string       // LastAckInfo human readable format of latest ack
+	ListenAddr     *net.UDPAddr // ListenAddr is the address of the listener
+	RemoteAddr     *net.UDPAddr // RemtoteAddr remote UDP address we send to.
+	RxAddr         *net.UDPAddr // RxAddr keeps track of what address we are sending to
+	LastRxAddr     *net.UDPAddr // LastRxAddr keeps track of what we last used
+	Quiet          bool         // Option to show output or not
+
+	printsPerSec   float64
+	packetsPerSec  float64
+	usecPerPkt     int32
+	usecPerPrint   int32
+	nextTxId       uint32
+	nextRxId       uint32
+	nextRxackId    uint32
+	startRtxtime   uint32 // remote's txtime at startup
+	startRxtime    uint32 // local rxtime at startup
+	lastRxtime     uint32 // local rxtime of last received packet
+	minCycleRxdiff int32  // smallest packet delay seen this cycle
+	nextCycle      uint32 // time when next cycle begins
+	now            uint32 // current time
+	nextSend       uint32 // time when we'll send next pkt
+	numLost        uint32 // number of rx packets not received
+	nextTxackIndex int    // next array item to fill in tx.acks
+	lastPrint      uint32 // time of last packet printout
+	latTx          int64
+	latTxMin       int64
+	latTxMax       int64
+	latTxCount     int64
+	latTxSum       int64
+	latTxVarSum    int64
+
+	latRx       int64
+	latRxMin    int64
+	latRxMax    int64
+	latRxCount  int64
+	latRxSum    int64
+	latRxVarSum int64
+}
+
+// Incremental standard deviation calculation, without needing to know the
+// mean in advance.  See:
+// http://mathcentral.uregina.ca/QQ/database/QQ.09.02/carlos1.html
+func onePassStddev(sumsq, sum, count int64) float64 {
+	numer := (count * sumsq) - (sum * sum)
+	denom := count * (count - 1)
+	return math.Sqrt(DIV(numer, denom))
+}
+
+// UsecMonoTimeNow returns the monotonic number of milliseconds since the program started.
```

Comment:
Comment says milliseconds but code returns microseconds. I think the code is right and the comment is wrong.